core/frontend/src/quantity-formatting/QuantityFormatter.ts

```diff
@@ -551,8 +554,16 @@ export class QuantityFormatter implements UnitsProvider {
   /** async method to set a units provider and reload caches */
   public async setUnitsProvider(unitsProvider: UnitsProvider) {
     this._unitsProvider = unitsProvider;
-    // force all cached data to be reinitialized
-    await IModelApp.quantityFormatter.onInitialized();
+
+    try {
+      // force all cached data to be reinitialized
+      await IModelApp.quantityFormatter.onInitialized();
+    } catch(_) {
+      Logger.logWarning(loggerCategory, "An exception occurred initializing the iModelApp.quantityFormatter with the given UnitsProvider. Defaulting back to the internal units provider.");
```

Comment:
This shouldn't log to the "IModelConnection" logger category, it has nothing to do with an IModelConnection.
Maybe just throw it in the "Package" category.

Reply:
Or I guess do what a handful of other places do when they don't want to create an official category for a one-off logging message:
`${FrontendLoggerCategory.Package}.quantityFormatter`.

Reply:
Thanks for the suggestion.  Done.